test/simdat.jl

```diff
@@ -39,3 +39,160 @@ using Test
     @test Tables.isrowtable(dat)
     @test Set(Tables.columnnames(first(dat))) == Set([:subj, :item, :dv])
 end
+
+@testset "simdat_crossed between-subjects between-items - simple case" begin
+    # stimulate data for a case in which a factor is both between-subject and between-item
+    conditions = Dict(:cond => ["A", "B"])
+    cond_n = length(conditions[:cond])
+    subj_n = 2
+    item_n = 4
+
+    data = DataFrame(simdat_crossed(subj_n, item_n,
+                                    subj_btwn = conditions,
+                                    item_btwn = conditions))
+
+    @test nrow(data) == subj_n * (item_n/cond_n)
+
+    # Test whether each subject is only in one of the conditions
+    for s in nlevels(subj_n,"S")
+        @test length(unique(data[isequal.(data.subj, s), :cond])) == 1
+    end
+
+    # Test whether each item is only in one of the conditions
+    for i in nlevels(item_n,"I")
+        @test length(unique(data[isequal.(data.item, i), :cond])) == 1
+    end
+
+end;
+
+@testset "simdat_crossed between-subjects between-items - complex case" begin
+    # stimulate data for a case in which a factor is both between-subject and between-item
+    subj_btwn = Dict(:age => ["O", "Y"],
+                     :cond1 => ["A", "B"],
+                     :cond2 => ["C", "D", "E"])
+    item_btwn = Dict(:pet => ["cat", "dog"],
+                     :cond1 => ["A", "B"],
+                     :cond2 => ["C", "D", "E"])
+    both_win = Dict(:time => ["morning", "evening"])
+
+    subj_n = 12
+    item_n = 12
+
+    data = DataFrame(simdat_crossed(subj_n, item_n,
+        subj_btwn = subj_btwn,
+        item_btwn = item_btwn,
+        both_win = both_win))
+
+    # Test whether each subject is only in one of the levels of the between-subject/between-items conditions
+    for s in nlevels(subj_n, "S"), cond in [:cond1, :cond2]
+        @test length(unique(data[isequal.(data.subj, s), cond])) == 1
+    end
+
+    # Test whether each item is only in one of the conditions
+    for i in nlevels(item_n, "I"), cond in [:cond1, :cond2]
+        @test length(unique(data[isequal.(data.item, i), cond])) == 1
+    end
+
+end;
+
+@testset "simdat_crossed test all combinations" begin
+        subj_btwn = Dict(:age => ["O", "Y"], # -> item within
+                     :cond1 => ["A", "B"],
+                     :cond2 => ["C", "D", "E"])
+    item_btwn = Dict(:pet => ["cat", "dog"], # -> subject within
+                     :cond1 => ["A", "B"],
+                     :cond2 => ["C", "D", "E"])
+    both_win = Dict(:time => ["morning", "evening"])
+
+
+    subj_n = 12
+    item_n = 12
+    data = DataFrame(simdat_crossed(subj_n, item_n,
+        subj_btwn=subj_btwn,
+        item_btwn=item_btwn,
+        both_win=both_win))
+    s2 = subset(data, :subj => x -> x .== "S02")    
+    @test all(s2.cond1 .== "A")
+    @test all(s2.age .== "Y")
```

Comment:
```suggestion
    @test all(==("A"), s2.cond1)
    @test all(==("Y"), s2.age)
```

`==("A")` is an example of [currying](https://en.wikipedia.org/wiki/Currying) / partial function application. Basically, it creates a new one argument function for testing equality against `"A"`: `x -> x == "A"`. Then we use the variant of `all` that takes a function as its first argument. Why do it this way? It avoids the intermediate allocation of an entire boolean vector. Moreover, it allows for short-circuiting -- as soon the first `false` occurs, it can stop computing the rest of the checks.